Feature/obs error assignment tool

azadeh-gh opened the pull request:
## Pull Request Description

This PR addresses the following changes:

1. **Fix Pycodestyle Violation**: Corrected a Pycodestyle violation in `/scripts/plot_proftrace.py` within the `develop` branch.
   
2. **New Feature Addition**: Added a new Python tool, `ObserrorAssignment.py`, to `ush/obsErrorAssignmentTool`, facilitating the automated estimation of observation error parameters.
3. **fix/cloudy_radiance_info.txt**: Needed to read `clw_clr`, and `clw_cld`

### Changes Included:

- **`obsErrorAssignmentTool` Directory Structure:**
  - `ObserrorAssignment.py`: Python tool for estimating observation error parameters.
  - `run_error_model_estimate.sh`: Shell script to invoke the Python script.
  - `README.md`: Provides an overview and usage instructions for the code.


### Functionality of `ObserrorAssignment.py`:

- Computes the mean cloud amount and first guess standard deviation.
- Generates four plots:
  - Standard deviation of first guess departures in each channel as a function of mean cloud amount: (clw_guess + clw_obs)/2.
  - Number of observations in each defined bin of mean cloud amount.
  - Histogram of un-normalized and normalized first guess departures in each channel.
  - Histogram of errors.
- Estimates error parameters (`error_cld`, `error_clr`, `clw_clr`, and `clw_cld`) and creates plots and CSV files for each channel.

### Example of output plot and CSV file:
![atms_n20_ch2](https://github.com/NOAA-EMC/PyGSI/assets/51101867/977d27ec-9a28-4293-9262-9f224d127f82)


![image](https://github.com/NOAA-EMC/PyGSI/assets/51101867/b24b2795-73e8-48ee-b696-d21005d1d27a)

azadeh-gh committed:
added source codes along with input and output example files

azadeh-gh committed:
Create README.md

azadeh-gh committed:
Update README.md

azadeh-gh committed:
Update README.md

azadeh-gh committed:
Updated some comments

azadeh-gh committed:
Merge branch 'feature/obsErrorAssignmentTool' of https://github.com/NOAA-EMC/PyGSI into feature/obsErrorAssignmentTool

azadeh-gh committed:
Update run_error_model_estimate.sh

azadeh-gh committed:
Update README.md

azadeh-gh committed:
Merge branch 'feature/obsErrorAssignmentTool' of https://github.com/NOAA-EMC/PyGSI into feature/obsErrorAssignmentTool

azadeh-gh committed:
Applied Linting

azadeh-gh committed:
Update README.md

azadeh-gh committed:
Update README.md

azadeh-gh committed:
Update README.md

azadeh-gh committed:
Fixed  pycodestyle violation already in develop

azadeh-gh committed:
Merge branch 'feature/obsErrorAssignmentTool' of https://github.com/NOAA-EMC/PyGSI into feature/obsErrorAssignmentTool

azadeh-gh committed:
Modified the ploting part of the python code

azadeh-gh committed:
Merge branch 'feature/obsErrorAssignmentTool' of https://github.com/NOAA-EMC/PyGSI into feature/obsErrorAssignmentTool

azadeh-gh committed:
fixed coding norms

kevindougherty-noaa commented on ush/obsErrorAssignmentTool/run_error_model_estimate.sh:
```diff
@@ -0,0 +1,49 @@
+
+# specify the path to the input diag files
+
+config_path=/scratch1/NCEPDEV/stmp4/Azadeh.Gholoubi/GDAS-ops/PyGSI/ush/obsErrorAssignmentTool/config_files/ 
```
I would make this more generic. Maybe just: `/PyGSI/ush/obsErrorAssignmentTool/config_files/ `

kevindougherty-noaa commented on ush/obsErrorAssignmentTool/run_error_model_estimate.sh:
```diff
@@ -0,0 +1,49 @@
+qc_flag=0
+
+# specify the path to saving output plots and csv file
+output=/scratch1/NCEPDEV/stmp4/Azadeh.Gholoubi/GDAS-ops/PyGSI/ush/obsErrorAssignmentTool/output/$bindir/
```
Same comment as above.

kevindougherty-noaa commented on ush/obsErrorAssignmentTool/run_error_model_estimate.sh:
```diff
@@ -0,0 +1,49 @@
+if [ $machine = orion ]; then
+   PyGSI=${PyGSI:-/Path/to/PyGSI/} # Change this to your own branch
+elif [ $machine = hera ]; then
+   PyGSI=${PyGSI:-/scratch1/NCEPDEV/stmp4/Azadeh.Gholoubi/GDAS-ops/PyGSI/} # Change this to your own branch
```
Same as above. Rather than your specific path, just more generic.

kevindougherty-noaa commented on ush/obsErrorAssignmentTool/ObsErrorAssignment.py:
```diff
@@ -0,0 +1,452 @@
+import argparse
```
Overall, looks great. Do you mind adding a short docstring in each function so future users know what the function is doing?

kevindougherty-noaa reviewed (commented):
Overall, looks great! I just have a few minor comments, but should be ready to go after those changes.

azadeh-gh committed:
ddressed reviewer comments. added docstring in each function and modified paths

azadeh-gh commented:
@kevindougherty-noaa , Thank you for your comments Kevin. I modified the paths, and added docstring in each function. Also changed some functions name to be more descriptive

kevindougherty-noaa reviewed (approved):
Looks great!
